pkg/steps/lease.go

```diff
@@ -107,6 +112,48 @@ func (s *leaseStep) run(ctx context.Context) error {
 	}
 }
 
+func acquireLeases(
```

Comment:
Technically, this could deadlock if we sort the leases in different orders and jobs acquire them in parallel but in separate orders. Can we sort by some identifier before acquiring?

Reply:
Unlikely and temporary, but I like the paranoia.  Updated.